### DISCUSSION TITLE
Question sur YaSolR

### DISCUSSION THREAD

**Discussion by mathieucarbou** [mathieucarbou/YaSolR#42] [General & Support]
> Bonjour Mathieu,
> 
> Félicitation pour ce routeur que je viens de découvrir via un sujet du forum HACF, et qui me semble très complet par rapport aux autres routeurs que j'ai essayé. Mon premier routeur était celui de Xlyric et mon routeur actuel, est celui d'André (F1ATB).
> 
> Je souhaiterais modifier mon installation sur une version Elite avec un JSY pour les mesures.
Cependant, ma maison possède un tableau électrique principal (mesure consommation maison) sur lequel est raccordé un tableau électrique secondaire de ma buanderie, où est raccordé mon chauffe eau et d'autres appareils.
Si je veux mettre un JSY, j'ai le problème que j'ai la mesure de la consommation de la maison est éloigné de mon chauffe eau pour le pilotage du SSR et la mesure de la charge du chauffe eau.
> 
> Il y a-t-il un moyen d'avoir le routeur à côté de mon chauffe eau et avoir la mesure de la consommation de la maison déporté avec un JSY ? (1 JSY pour la maison et un JSY pour le chauffe eau ?)
> 
> J'ai vu aussi qu'il est possible de vous commander un module de détection de passage à zéro
(4 en stock). Suivant la faisabilité des mesures de ma question précédente, comment faire pour le commander.
> 
> Cordialement,
> Olivier R.

Bonjour,

Merci!

Alors oui, YaSolR est compatible avec un JSY installé en remote. 
YasolR supporte presque tous les JSY, donc il suffit d'installer dans le tableau un ESP32 avec un JSY (JSY-MK-163T par exemple) et l'appli Mycila JSY UDP sender: https://mathieu.carbou.me/MycilaJSY/#remote-jsy.

Voir le manuel de YaSolR pour le remote: https://yasolr.carbou.me/manual#jsy-remote

Cette appli est indépendante de YaSolR et expose une API REST qui peut aussi être interrogée pour obtenir les mesures. 
Elle détecte automatiquement le JSY qui est branché, le bascule dans sa vitesse la plus rapide, et affiche toutes les stats, selon le JSY (1 channel, 2 channel, ou tri), et envoie les stats par UDP sur le réseau.
L'avantage de ce mode est qu'il est aussi rapide que d'avoir un JSY connecté à même l'ESP car le taux de transfert des mesures par UDP est très élevé.
Et l'UDP permet également à n'importe qui de recevoir les données, sans avoir besoin de configuration.

Pour mesurer la puissance routée, ll faudra utiliser un PZEM.

En résumé:

- pour la mesure de la sortie (local) - https://yasolr.carbou.me/#measurement-device-optional

  * JSY-MK-193 first channel (AC, RS485 interface)
  * JSY-MK-194T and JSY-MK-194G first channel (AC, TTL interface)
  * PZEM-004T V3

- pour la mesure Enedis (local) - https://yasolr.carbou.me/#grid-power-measurement

  * JSY-MK-163T (AC, TTL interface)
  * JSY-MK-193 second channel (AC, RS485 interface)
  * JSY-MK-194T and JSY-MK-194G second channel (AC, TTL interface)
  * JSY-MK-227 and JSY-MK-229 (AC and DC, RS485 interface)
  * JSY-MK-333 and JSY-MK-333G (AC, RS485 interface)

- pour la mesure Enedis (remote) - https://yasolr.carbou.me/#grid-power-measurement

  * MQTT (Shelly EM, 3EM, etc)
  * Remote JSY

Je n'ai encore jamais essayé d'utiliser un JSY remote pour Enedis + un JSY local pour les sorties, mais le code le supporte. 
Les raisons:

-  le channel 1 du JSY est utilisé pour mesurer l'ensemble des sorties et non chaque sortie, comme les PZEM, car les PZEM peuvent fonctionner en cascade via un adressage. Donc chaque PZEM est assigné à une sortie et peut mesurer chaque sortie de manière indépendante. Quand on utilise un JSY channel 1 pour la mesure, le device ne sait pas s'il mesure une ou deux sorties...

- un JSY coûte beaucoup plus cher qu'un PZEM

- pour mesurer une sortie, pas besoin d'avoir un device bi-directionnel comme un JSY (la bi-directionnalité est requise uniquement pour Enedis).

- la mesure des sorties n'est pas requise au routage, c'est optionnel et uniquement pour les stats.

C'est important de comprendre aussi que la mesure Enedis peut venir d'un Shelly par MQTT: si vous avez déjà un Shely EM, alors vous avez déjà 2 pinces pour mesurer Enedis + sortie, donc inutile de rajouter d'autres composants (à par pour obtenir une meilleur précision de routage)

Si vous n'avez rien acheté encore, je suggère:
- 1x JSY-MK-163T ou JSY-MK-194G (modèle avec 2 pinces) dans le tableau pour mesurer Enedis
- 2x ESP32 NodeMCU Dev Kit
- Optionnel: un PZEM pour mesurer la sortie
- pour dimmer:
  * soit SSR Random LCTC + ZCD
  * soit LSA + module de contrôle

Note: le mieux est d'utiliser un LSA, il est supporté actuellement mais requiert d'alimenter un module de contrôle via un MeanWell HDR-15-12 en 12V DC, ce qui est contraignant. Je vais ajouter bientôt (d'ici 1 mois je pense) le support d'un autre module (un DAC), qui va permettre de contrôler directement le LSA via un ESP32 et sa tension de 3.3V, avec une précision digitale de 12 bits. Donc il sera bientôt possible de faire un routeur uniquement avec ESP32 + DAC + LSA. Le LSA est beaucoup plus costaud que le SSR (3 fois plus gros), et contient un module ZCD. C'est le top en régulation de tension. Donc le support du DAC pour alimenter le LSA via l'ESP va grandement faciliter son utilisation.

Il me reste des modules ZCD (https://yasolr.carbou.me/pro), avec ou sans support DIN. 
Il faut rajouter les frais postaux en plus (quelques euros).

Cdlt,

Mathieu.

**Comment by florentbr** [comment]
@mathieucarbou

> Note: le mieux est d'utiliser un LSA, il est supporté actuellement mais requiert d'alimenter un module de contrôle via un MeanWell HDR-15-12 en 12V DC, ce qui est contraignant. Je vais ajouter bientôt (d'ici 1 mois je pense) le support d'un autre module (un DAC), qui va permettre de contrôler directement le LSA via un ESP32 et sa tension de 3.3V, avec une précision digitale de 12 bits. Donc il sera bientôt possible de faire un routeur uniquement avec ESP32 + DAC + LSA. Le LSA est beaucoup plus costaud que le SSR (3 fois plus gros), et contient un module ZCD. C'est le top en régulation de tension. Donc le support du DAC pour alimenter le LSA via l'ESP va grandement faciliter son utilisation.

La datasheet du LSA-H3P50YB indique qu'il fonctionne sans alimentation externe et qu'il a une entrée PWM.
Il devrait donc être facilement pilotable avec `analogWrite` ou LEDC à 1-3kHz en variant le duty-cycle.
Les sorties de l'ESP sont en 3v. Le LSA requière 4v minium. Il faudra donc interfacer un [level-shifter](https://i.ebayimg.com/images/g/lF0AAOSwpRRWoLVn/s-l500.webp)  ou un module [opto-isolator](https://i.ebayimg.com/images/g/N7cAAOSweoBbFj10/s-l500.webp)  avec l’entrée PWM et la sortie 5v du LSA.

**Reply by mathieucarbou** [reply]
En fait il faut une alimentation externe si ok utilise un module pwm => 0-10V analogique qui puisse convertir le duty cycle envoyé au triac en tension pour une des entrées du LSA. J'ai testé cette option pour l'instant, qui permet de réutiliser exactement le même code côté ESP car au lieu de mettre un Robodyn ou SSR random on peut mettre ce convertisseur analogique.

J'ai essayé avant cette option avec un level shifter mais ce n'était pas concluant, Seb de PVBrain a aussi essayé et c'est finalement plus compliqué.

Le plus simple reste l'utilisation d'un DAC 12-15 bits Dfrobot qui permet une précision assez grande tout en ayant la possibilité de booster jusqu'à 10V le 3.3V de l'esp. Seb l'a implémenté dans PVBrain sur ESPhome, et dans le cas de YaSolR, ça permettrait de réutiliser partiellement la lib de dimmer, moyennant le fait que l'utilisateur déclare quel type de dimmer il utilise au final.

Perso, j'aime la simplicité du code et je trouve la solution pwm-analogique plus simple à mettre en place : le module n' est pas cher, le code est le même, il suffit d'une alimentation 12V via un meanwell hdr-15-12 par exemple... Le DAC complexifie pas mal le code car je dois supporter 2 libraires supplémentaires. Et le level shifter je ne suis pas arrivé à la faire fonctionner, Seb pas non plus de façon concluante, je me rappelle aussi qu'il disait qu'il y avait aussi un problème lié au fait que le LSA n'est pas linéaire.

**Reply by florentbr** [reply]
> Et le level shifter je ne suis pas arrivé à la faire fonctionner, Seb pas non plus de façon concluante, je me rappelle aussi qu'il disait qu'il y avait aussi un problème lié au fait que le LSA n'est pas linéaire.

C'est étonnant, l’entrée PWM est supposée être plus simple et plus precise. Soit le MCU du LSA converti en interne le PWM pour acquisition par ADC plutôt que par interruption, ou bien l’entrée est de type "current sink" plutôt que "current source". 
Une entrée de type "current sink" n'est pas adaptée pour un "level shifter" puisque sa pull-up pourra difficilement fournir le niveau logic haut (4v min). Que ce soit une entrée "current source" ou "current sink", un simple optoisolateur doit être suffisant pour piloter le module depuis un ESP, mais le câblage est different (5v/PWM  vs PWM/Gnd). Dommage qu'il n'y ait pas plus d'info sur ce module sur sa plage/linéarité/resolution.

**Reply by mathieucarbou** [reply]
Oui! Après, je ne m'y connais pas assez en élec pour pousser plus loin, je sais juste que d'autres ont essayé, la conclusion étant que passer par GP8403 (DfRobot) était plus simple (je parle niveau code aussi).
Il n'en reste pas moins que le LSA n'est pas linéaire sur son entrée 0-10V: il démarre à 1-1.5V et est à 100% autour de 8-8.5V. 
Du coup, bien que le GP8403 facilite le montage car pas d'alimentation secondaire nécessaire, le duty envoyé ne correspondra pas du coup à ce que le LSA fait. Le même problème existe avec le module pwm-analogique que je supporte actuellement et a besoin d'une tension 12V, mais moindre, car il y a des potentiomètres pour ajuster sur le module et sur le Meanwell.
Également, dans ma librairie dimmer, j'ai ajouté une fonction de remapping, comme les Shelly dimmer, afin de redéfinir le 0% (qui devient vers 10% et le 100% qui devient alors vers 80%. 
J'ai de bons résultats avec cette méthode, c'est ce que je fais aussi sur le routeur que j'ai conçu basé sur Shelly EM Pro (remapping via le shelly dimmer gen3).

**Reply by mathieucarbou** [reply]
L'entrée du LSA est de type current sink oui car il ne sort aucune tension mais lit un signal entrant. Il faut juste brancher le GND et ensuite envoyer soit un signal +5V pwm, soit un courant variable, soit une tension entre 0-10V.

**Comment by olivr2s** [comment]
Wahou....quelle réponse claire, précise et très technique....j'adore.
A l'aide de votre site et ces dernières informations, j'y vois beaucoup plus clair et même si mon chauffe est éloigné du tableau électrique, la version JSY remote permet de palier au problème avec une précision de 20 mesures / seconde.

J'ai bien compris que si je veux un routeur précis et réactif, ce sera avec mesure JSY et pilotage LSA. Et peut être dans quelques temps la possibilité de se passer du PWM et de l'alimentation 12V via le convertisseur DAC. Ce sera effectivement le top, un peu comme la version Shelly Dimmer mais avec toute la puissance du routeur YaSolR sur ESP32.
Je vais donc patienter et suivre de près cette évolution avec le DAC...

Par contre il me reste une interrogation concernant le ZCD. Sachant que le LSA possède un ZCD, il n'y a pas besoin de JSY avec ZCD (JSY-MK-194G), sinon ça fait doublon....

**Reply by mathieucarbou** [reply]
Oui exact, le LSA inclue son propre ZCD. Du coup, pour la tableau, un simple jsy-mk-163t ttl bi-directionnel suffit. À voir après si la différence de prix vaut le coup de se restreindre en évolutions possibles ?

**Reply by olivr2s** [reply]
Effectivement, suivant la différence de prix, autant prévoir un modèle l'ayant déjà au cas où d'une autre utilisation.
Je voulais être sûr que si les 2 possèdent un ZCD, que cela ne pose pas de problème au système.

**Reply by mathieucarbou** [reply]
Note: les lectures d'un jsy se font à un rythme d'environ 20-25 par seconde mais le jsy faisant une moyenne glissante sur une seconde et rendant disponible ses changements toutes les 300-330ms, on a en fait environ 3 changements sur lesquels reagir par seconde.

**Reply by mathieucarbou** [reply]
> Effectivement, suivant la différence de prix, autant prévoir un modèle l'ayant déjà au cas où d'une autre utilisation. Je voulais être sûr que si les 2 possèdent un ZCD, que cela ne pose pas de problème au système.

En fait il ne faut pas connecter le pin Zx du jsy-mk-194g à l'esp c'est tout car pas besoin ;-)

**Reply by olivr2s** [reply]
Merci, tout est clair maintenant.

**Comment by olivr2s** [comment]
Bonjour @mathieucarbou,

Comme j'ai reçu mon JSY-MK-194G, j'ai commencé le montage de mon routeur et du JSY Remote.
Pour le JSY-Remote, le montage, l'installation et la configuration ce sont très bien passé et j'ai bien les informations qui remontent.
(Utilisation que du CT1 pour les valeurs Enedis)
![JSY Remote](https://github.com/user-attachments/assets/373b94f4-d990-46b8-948e-4e86ec5be7fe)

Je suis passé ensuite sur le routeur avec la dernière version V1.0.0-rc8 qui vient tout juste de sortir sur un ESP32, et après installation, configuration, j'ai activé le JSY Remote dans Hardware, redémarré le routeur.

Quand je vais dans statistics, j'ai bien le nombre de messages par secondes qui remonte, la fréquence et la tension du réseau aussi.
![statistics](https://github.com/user-attachments/assets/c7e62413-d7a2-4903-8bc3-7525d4bcd5ac)

Par contre, lorsque je vais sur le dashboard, j'ai uniquement la tension et la fréquence, mais pas de puissance, le graphique est à zéro, et j'ai une puissance routée alors que c'est le cumul de puissance mesurée du JSY remote depuis sa mise en service....
![Overview](https://github.com/user-attachments/assets/5a5d70ed-d444-46f8-b121-6d6c3dce1e35)

Est-ce que j'aurais loupé un truc ?

**Comment by mathieucarbou** [comment]
@olivr2s : très simple! Il faut utiliser le CT2 pour Enedis. La raison est qu'il y a plusieurs modèles de JSY, et le channel 2 est toujours une pince alors que le channel 1 est un tore sur la carte des fois, donc utilisation possible uniquement pour mesurer l'ensemble des sorties ;-)

Je vais ajouter cette précisions à plusieurs endroits pour être sûr de ne pas la rater.

**Reply by olivr2s** [reply]
Effectivement, c'est beaucoup mieux ainsi, je vais pouvoir poursuivre le montage de mon routeur.
Encore merci pour tes réponses rapides.

**Reply by mathieucarbou** [reply]
De rien! Bon montage 😄

**Comment by olivr2s** [comment]
@mathieucarbou
En continuant le montage de mon routeur, j'ai rencontré des difficultés avec le Robodyn 40A que j'utilise avec mon routeur F1ATB.
Lorsque j'activais dans la partie Matériel le pilotage par variateur et après reboot, ça restait sur non activé / erreur.
Comme j'avais déjà rapidement testé il y a quelques jours le pilotage et que ça fonctionnait, je comprenais pas pourquoi cette erreur, si ce n'est que j'avais fait un câblage cette fois si plus propre et définitif.
J'ai contrôlé plusieurs fois mon raccordement, fait des tests de continuité....rien.

Après, l'autre différence par rapport à la dernière fois c'était la version du firmware (v1.0.0-rc7 contre maintenant rc8)
J'ai bataillé un moment, puis dans la partie matériel, j'ai activé "Débogage" et fait reboot, et du coup, plus de problème dans la partie matériel, le menu sortie 1 apparaît, et j'arrive à piloter le Robodyn. Par contre, si je désactive le "Débogage", de nouveau en erreur....

Comme la partie "Débogage" et maintenant dans le menu matériel depuis la version rc8, je voulais savoir si il faut le garder activé en permanence, ou c'est un problème dans la version rc7

**Reply by mathieucarbou** [reply]
Oui j'ai mis à jour la doc et indiqué dans le release note de la RC8 (https://github.com/mathieucarbou/YaSolR/releases/tag/v1.0.0-rc8) que dorénavant, tout changement (activation et désactivation) de matériel requiert un restart pour être pris en compte. C'est dans l'optique d'optimiser la mémoire en n'activant au démarrage uniquement les fonctions requises.

L'option de Debug n'est pas gourmande (je la laisse toujours activer) et permet de suivre ce que le routeur fait, mais n'est pas nécessaire en tant normal.

C'est l'option real-time dans le PID qui elle est gourmande et n'est pas à laisser active ;-)

**Reply by mathieucarbou** [reply]
Alors effectivement, si aucune tension n'est détectée, le dimmer ne démarre pas. 
Je vais fixer ça, pour qu'il démarre une fois une tension détectée, après le démarrage.
si une tension est présente au démarrage, il démarrera.

**Reply by olivr2s** [reply]
Oui, j'ai bien vu dans la documentation que maintenant toute modification dans matériel nécessite un reboot pour valider la modification, de ce côté aucun soucis.
Le côté consommation du Debug, ok pour moi et oui, le PID en temps réel est plus consommateur, bien noté.
Reste effectivement la non activation du variateur si je désactive le débogage dans le menu matériel, donc je le laisse actif pour le moment.

Petite question matériel.
J'envisage de remplacer mon Robodyn 40A avec son ZCD pas terrible et se voit de temps en temps sur les courbes, par un variateur LSA qui possède un ZCD correct.
Par contre cela nécessite un convertisseur PWM et une alimentation 12V en plus. J'ai vu dans les projets que le support pour le DAC GP8403 / GP8413 pour ne plus avoir besoin de l'alimentation 12V était en cours. A commander du matériel chez Aliexpress avec les délais de livraison, vaut mieux que je commande LSA+convertisseur PWM+alim 12V, ou LSA+GP8413 ou encore attendre....

**Reply by mathieucarbou** [reply]
`v1.0.0-rc9` is currently building and will be available soon. 

Just to re-explain how it works:
- dimmers require a ZCD module to be there
- dimmers will only start once ZCD analysis is done and a frequency is detected

So even if they are activated in hardware, they won't be started until electricity (and ZCD) are both active / there.

This will change in the future when these will be supported:
- LSA with 12-15 bits DAC such as DFRobot (no ZCD will be required - only ESP32 + DAC + LSA)
- ZC Sync dimmers (which only activates at ZC point), which will only support full wave modulation (less precise than triac or random ssr)

At that point, there will be a new config dropdown to specify which dimmer type is used, default one being `Triac / Random SSR`

**Reply by olivr2s** [reply]
J'ai mis à jour avec la version rc9, et maintenant plus de problème d'activation du variateur lorsque le débogage est désactivé.
MERCI.

**Reply by mathieucarbou** [reply]
> Par contre cela nécessite un convertisseur PWM et une alimentation 12V en plus. J'ai vu dans les projets que le support pour le DAC GP8403 / GP8413 pour ne plus avoir besoin de l'alimentation 12V était en cours. A commander du matériel chez Aliexpress avec les délais de livraison, vaut mieux que je commande LSA+convertisseur PWM+alim 12V, ou LSA+GP8413 ou encore attendre....

Vous pouvez attendre un peu pour ça ;-) Avec le Robodyn, ça fonctionnera déjà pas mal car l'algo actuel a des safeguards pour limiter les effets du mauvais ZCD du Robodyn. Ça sera définitivement mieux avec un SSR Random + ZCD, ou juste un LSA.

Pour le LSA, actuellement, il est supporté via le PWM-Analog converter avec une entée 12V supplémentaire...  C'est pas top donc car requiert cette alim en plus du module. Quand le support du DAC sera fait, je supprimerai cette possibilité car c'est bancale comme solution. Donc je dirai qu'il vaut mieux attendre ;-)

Je vais supporter le GP8403 (DFR0971) sur 12 bits et si une détection de module est possible je supporterai aussi ces deux là sur 15 bits:

- GP8413 (DFR1073)
- GP8211S (DFR1071)

https://www.dfrobot.com/blog-13458.html

**Comment by nick-GBC** [comment]
Hi Mathieu
Forgive me using English as opposed to French. my French is not up to a technical discussion!

I am in the UK and wanting to try what looks like an excellent PV Router. I already have a commercial version which has died twice and does not have the features i want. 

We have 32Kw of solar on the shed roof and i want to maximise it!!

I want to go down what you describe as the'Professional' route....

So is there an LSA which you prefer - I have 3Kw immersion. I have a  JSY-MK-194G on order and already have ESP32's and a PZEM.

I need to get a bypass relay for full diversion when our Octopus power rates go negative - nice to use power when they a paying me :)
I have determined that I can switch on the bypass relay from Home Assistant with an api POST when my Octopus integration tells me the electricity rate is negative or zero.

Last question - is there a possibility of having a the second output as a remote dimmer on a second ESP32 ? - i have a 2nd 3KW immersion in the biomass buffer tank in the garage :)

Many thanks 
Regards
Nick

**Reply by mathieucarbou** [reply]
> is there a possibility of having a the second output as a remote dimmer on a second ESP32 ? - i have a 2nd 3KW immersion in the biomass buffer tank in the garage :)

Hello,
No problem for English! 
Any language is fine as long as it is FR or EN for me lol ;-)

So YaSolR can handle up to 2 outputs for solar excess diversion, but only the measurement part can be placed remotely in the electrical box (with the JSY UDP Remote app). Dimmer and relay control is currently all done locally connected to the board. In the future I will implement remote modes with ESP-Now, but this won't be soon. The trick is that the ESP-dimmer com has to be reliable and fast otherwise the routing loop could be impacted. To implement remote mode correctly this is not as trivial as it might seem to reduce the impact of delayed data on the diversion.

So if you really need to have 2 different locations, which cannot be controlled from the same location, then for now you can setup 2 YaSolR routers, except that the second one will take its grid power data from MQTT, and it will be the topic `/<yasolr-root>/router/virtual_grid_power`, which is the grid power if the first yasolr router was not diverting. The second router wll then have a priority. It is also possible to set limits on routeurs: like only divert a max of 500W, or 50% of the dimmer duty, etc.

Right now, mqtt publications happen by default each 5 sec, they could be reduced if the wifi connection is good. 
But of course even if reduced, the divert precision will suffer compared to having everything locally connected to the ESP32.

About the LSA: this is a good choice for long term.
Even if right now it can only be controlled with a 12V input and still requires a ZCD, I am working on adding support for LSA without the need of a ZCD, and also without the need of an alternative 12V input.

**Comment by mathieucarbou** [comment]
@nick-GBC @olivr2s : fyi I have updatd YaSolR website:
- https://yasolr.carbou.me/build#the-professional
- https://yasolr.carbou.me/#dimmer-required
to add more info about what is implemented, what is coming and the possible combinations for the LSA with YaSolR.
I am working at the moment in the PWM and DAC support. Now, controlling the LSA requires a ZCD module and an external 12V plus a PWN-Analog converter because it uses the same dimmer code as for a random ssr or robodyn.

**Reply by nick-GBC** [reply]
Hi
Many thanks for your prompt reply and the updates. I will order more parts and start building. 
It may take a while, my eyesight isn't what it was when it comes to soldering header pins etc. - human evolution has not kept up with the production of smaller and smaller devices :(

Thanks and regards

Nick

**Reply by nick-GBC** [reply]
Hi Mathieu

I am testing the router/dimmer at the moment using the latest build. I accidentally ordered the wrong DAC. 

https://www.dfrobot.com/product-2752.html?srsltid=AfmBOopHrHTtmXk2g8cW0YHQ2e3ohddCER3ZIQXUZsvt145_LnDNGRne

It does seem to work using PWM to 5V using the 5V setting to power an LSA,I am testing at the moment with a small oil radiator as the load. I am using the JSY-MK-194G to provide the load readings and ZCD to the ESP32 and getting the grid power/voltage via MQTT from Home Assistant.

The dimmer is working in manual mode but not in auto, the graph bottom left shows grid power but the overview above does not.
Any idea where I should look to fix this ! ?

[doc1234.docx](https://github.com/user-attachments/files/18785547/doc1234.docx)

I have re-ordered a DAC having found this one, which I assume is the same as the DAC your reference

https://thepihut.com/products/dac-2-i2c-unit-gp8413?srsltid=AfmBOoochV38s_-G79EBjx4oSRUkvj5a3S0Jv136-0Twd8ZAj1frwHAA

I also need to get a bigger LSA as I stupidly order a 10A one instaed of 40A...

If I switch to the PRO version do I get the same latest versions as those on your release site, the trial version is at V8 by the look of it :)

Regards

Nick

**Reply by mathieucarbou** [reply]
HI @nick-GBC : DAC will be supported soon. All the things marked with 🚧 in the website are "under construction" ;-)
I plan to support:
- LSA / LCTC Voltage Regulators + DAC GP8211S (DFR1071) (🚧)
- LSA / LCTC Voltage Regulators + DAC GP8403 (DFR0971) (🚧)
- LSA / LCTC Voltage Regulators + DAC GP8413 (DFR1073) (🚧)
The code will be implemented soon.

**Comment by olivr2s** [comment]
Bonjour @mathieucarbou ,

Mon routeur tourne bien depuis quelques jours et c'est top.
J'ai bien tous les informations et commandes qui remontent aussi dans Home Assistant et je vois bien les modifications du routeur remonter vers HA.
Par contre si je veux modifier ou commander le routeur par HA, il ne se passe rien....
Une idée de pourquoi de HA vers le routeur cela ne prend pas en compte ?

**Reply by mathieucarbou** [reply]
Bonjour,

Si le routage automatique est activé, ce n'est pas possible de contrôler manuellement.
Il faut désactiver le routage auto pour controller manuellement le dimmer.

Idem pour le bypass.

<img width="310" alt="image" src="https://github.com/user-attachments/assets/5cafdb02-2448-4b3a-a554-1b61ccc5b289" />

**Reply by olivr2s** [reply]
Désolé, j'ai oublié de préciser que j'avais bien désactivé les modes automatique pour essayer de changer des valeurs.
D'ailleurs on voit bien dans le dashboard du routeur que les commandes apparaissent/disparaissent en fonction du mode automatique actif/inactif, ce qui ne se voit pas dans HA.
D'ailleurs il faudrait que je mette en place une condition d'apparition des commandes en fonction du mode automatique dans HA pour enlever toute ambiguïté.

Mais mon problème reste. Si je veux désactiver le routage automatique dans HA, le bouton bascule et revient actif quelques secondes après et dans le dashboard du routeur il ne se passe rien....
Et si le routage automatique est désactivé à partir du dashboard du routeur, quand je pilote le routage avec le curseur dans HA, la valeur reste, mais il ne se passe rien au niveau du routeur.

**Reply by mathieucarbou** [reply]
Oui dans HA je ne peux pas jouer avec la visibilité des composants facilement... Je vais regarder! J'avais testé cette semaine et ça fonctionnait, mais j'ai fait pas mal de MAJ dans le code dernièrement pour commencer le support des dimmer PWM et DAC

Autre note: les publication MQTT  se font aux 5 secondes, donc il faut attendre au max 5 secondes pour avoir un feedback d'une action depuis HA (feedback étant par exemple voir les valeurs de routage suite à déplacement du slider)

**Reply by mathieucarbou** [reply]
je peux reproduire - je fixe ça...

**Reply by mathieucarbou** [reply]
Le bug est corrigé et la nouvelle version v1.0.0-rc10 devrait être dispo d'ici 30min.

Il y a eu des changements en préparation des différents types de dimmer supportés, donc bien sélectionner le bon type de dimmer dans la page hardware.

<img width="722" alt="image" src="https://github.com/user-attachments/assets/3e509816-d3ef-41fd-93e6-ee69d1baf113" />

**Reply by olivr2s** [reply]
Je testerais une fois rentré du boulot.
Merci pour la précision pour la configuration du dimmer dans la partie hardware.
